### PR TITLE
AWS Lambda Provider

### PR DESCRIPTION
**Pull request by wtcross**
This PR introduces a provider for [AWS Lambda](http://aws.amazon.com/lambda/).

Right now you have to either upload a zip file in the UI, through the command line, or use their console editor in the browser to create a Lambda. Lambda's are just JavaScript, so I want to be able to use the pull request development flow on GitHub with TravisCI for CD of my Lambdas. I think that being able to continuously deploy Lambdas from TravisCI could help grow an ecosystem of community Lambdas. At least that's an idea.. :)

The `lambda` provider is functional in its current state and has 100% test coverage. That said, there might be some issues that we should discuss if you even want to include it in the first place. Please let me know if there are any fundamental concerns with exposing this provider. Lambda is still in preview, but wanted to get the conversation started ahead of time.

### CODE REVIEW

**Comment by wtcross** [comment]
Any thoughts on this provider?

File: Gemfile [hunk 1]
```diff
@@ -31,6 +31,11 @@ group :code_deploy do
   gem 'aws-sdk', '2.0.13.pre'
 end
 
+group :lambda do
+  gem 'aws-sdk', '2.0.13.pre'
```
Comment (BanzaiMan) [inline Gemfile]: Is it possible to specify any 2.0.x.pre here? The latest as of this writing is 2.0.18.pre.

**Comment by wtcross** [comment]
I updated the version specification in the Gemfile as well as provider. Thank you for the quick response!

**Comment by wtcross** [comment]
This provider is going to possibly be one of the easiest ways to perform CI/CD for an AWS Lambda. Devs will get the full pipeline for free with open source projects (GitHub + TravisCI + AWS Lambda free tier). It will essentially let devs treat a Lambda as a regular NodeJS project, giving them access to all NPM packages. Could make sharing and reusing Lambda's incredibly easy.

That said, Lambda is still in preview. I don't know of the official 'ready' date.

I welcome any feedback on preparing this provider for when Lambda is no longer in preview. If this PR is still in consideration I'd like to fix any problems or make any enhancements that are suggested. Feel free to be super critical.

**Comment by BanzaiMan** [comment]
Sorry for the delay. Please test it with `edge: true`. Thank you.

**Pull request merged**